woocommerce/woocommerce
# Use AJAX for Page Configuration

### All Submissions:


### Changes proposed in this Pull Request:

Since the page selection dropdowns in the Advanced settings area are populated on load we can end up taking a very long time to load on shops that have lots of pages. This PR aims to replace the existing dropdown with search selects that will reduce the page load time by pulling the pages using AJAX.

Closes #27233.

### How to test the changes in this Pull Request:

1. Review `WooCommerce > Settings > Advanced`
2. Ensure page dropdowns are searchable selects.
3. Ensure you can clear the currently selected page.
4. Ensure you can only select one page at a time.
5. Ensure excludes are working correctly. For example cart page should not show checkout page as available. And checkout page should not show cart page as available.
6. Ensure you see the page ID next to the page name.
7. Ensure page dropdowns still save correctly
8. Go to the frontend and ensure these pages are indeed set correctly by loading them.

### Other information:




### Changelog entry

> Change - replace page setup dropdowns with AJAX powered search selects.
